### PR TITLE
Factor out some Travis code

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

Lots of redundancies in `.travis.yml` that make it difficult to evolve.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: ci/travis/install-bazel.sh [hunk 1]
```diff
@@ -50,21 +50,15 @@ else
   target="./install.sh"
   curl -s -L -R -o "${target}" "https://github.com/bazelbuild/bazel/releases/download/${version}/bazel-${version}-installer-${platform}-${achitecture}.sh"
   chmod +x "${target}"
-  "${target}" --user > /dev/null
+  if [ "${TRAVIS-}" = true ] || [ -n "${GITHUB_WORKFLOW-}" ]; then
+    sudo "${target}" > /dev/null  # system-wide install for CI
+    command -V bazel 1>&2
```
Comment (simon-mo) [inline ci/travis/install-bazel.sh:55]: what does this do?

**Review by simon-mo — APPROVED** [review]
LGTM. The other question is that if the output will be hidden anyway, why are we piping some output to /dev/null? (e.g. bazel installation script)

File: ci/travis/install-bazel.sh (hunk 1 above)
Comment (mehrdadn) [reply ci/travis/install-bazel.sh:55]: It prints something like `bazel is /home/username/bin/bazel` to stderr, that's all.

**Comment by mehrdadn** [comment]
@simon-mo Oh I don't think I'm hiding `install-bazel.sh`'s output, at least not in GitHub Actions. And it can have a couple lines of output aside from the one from the installer which is useful for knowing that the installer ran etc.

**Pull request merged**